OpenRA.Mods.Common/Traits/Harvester.cs

```diff
@@ -243,21 +161,29 @@ public void AcceptResource(Actor self, string resourceType)
 			UpdateCondition(self);
 		}
 
-		// Returns true when unloading is complete
-		public virtual bool TickUnload(Actor self, Actor proc)
+		IAcceptResources acceptResources;
+		public override void DockStarted(Actor self, Actor hostActor, DockHost host)
+		{
+			if (DockingPossible(host.GetDockType))
+				acceptResources = hostActor.TraitOrDefault<IAcceptResources>();
+		}
+
+		public override bool DockTick(Actor self, Actor hostActor, DockHost host)
 		{
+			if (acceptResources == null || IsTraitDisabled)
+				return true;
+
 			// Wait until the next bale is ready
 			if (--currentUnloadTicks > 0)
 				return false;
 
 			if (contents.Keys.Count > 0)
 			{
-				var acceptResources = proc.Trait<IAcceptResources>();
 				foreach (var c in contents)
 				{
 					var resourceType = c.Key;
```

Comment:
As per the prior comment, here this should either be `contentsType`, or the above `contents` dictionary should be renamed to `resources`.

Reply:
answered the in above comment